### DISCUSSION TITLE
how to specify rendering "clean"?

### DISCUSSION THREAD

**Discussion by ijlyttle** [quarto-dev/quarto-cli#3819] [Q&A]
This is related to, but distinct from, #3199. 

I understand that I can specify the output directory to be deleted before rendering by running:

```
quarto render --clean
```

I also see that this is default when the render-flags are parsed: 

https://github.com/quarto-dev/quarto-cli/blob/1c4b8b10075e095a4671c00274064ff9b82b3b51/src/command/render/flags.ts#L346-L349

However, if I run:

```
quarto publish
```

it calls render (unless `--no-render`), but the `clean = true` default is not invoked as far as I can tell.

## Questions

- Is there a way to specify  a `clean:` value in `_quarto.yml`?

- Rather than setting `clean = true` in `flags.ts`, would you consider a change in the render logic?

  https://github.com/quarto-dev/quarto-cli/blob/1c4b8b10075e095a4671c00274064ff9b82b3b51/src/command/render/project.ts#L161-L164
  
  I think that

  ```ts
  (options.flags?.clean !== false)
  ```

  would have the same effect as setting the default value in `flags.ts`, but it would work also in the `publish` case. 

Happy to create a PR, etc. (Happy New Year 🎉)

**Comment by cscheid** [comment]
Just thinking this through, that check is using the "equality" operator `==` instead of the "strict equality" operator `===`:

```
$ deno
> 1 == true
true
```

What values are you thinking would be passed to `clean` that `!== false` would be better than `== true`? Note that

```
> 0 !== false
true
> null !== false
true
> undefined !== false
true
```

So I think you're thinking of something else.

**Reply by ijlyttle** [reply]
Basically, I propose`undefined` to be functionally equivalent to `true`, as set in `flags.ts`:

```ts
 if (flags.clean === undefined) { 
   flags.clean = true; 
 } 
```

Along those lines, what if this bit of code were to be put into `project.ts`, before the conditional?

**Reply by cscheid** [reply]
I see. That will actually change our default behavior, and I don't know what expectations already exist in the ecosystem. I'll bring this up in our weekly meetings and get back to you.

**Reply by ijlyttle** [reply]
Fair enough - FWIW, if the default for `publish` remains the same, having a way to specify render-flags in `_quarto.yml` could be a way around it (whether it's a good idea is another question).

Thanks!
